R/correctionsataglance-data.R

```diff
@@ -1,357 +1,539 @@
+#' Parse CORR Approvals
+#'
+#' @description Reads and formats the approvals for use on the CORR report
+#' @param timeSeries The time series to get approvals from
+#' @param timezone The timezone to parse data into
+#' @param dateSeq The sequence of dates from the report start date to the report end date
+parseCorrApprovals <- function(timeSeries, timezone, dateSeq){
+  approvals <- timeSeries[['approvals']]
+  returnData <- list()
+  
+  if(!isEmptyOrBlank(approvals)){
+    approvals[['startTime']] <- flexibleTimeParse(approvals[['startTime']], timezone)
+    approvals[['endTime']] <- flexibleTimeParse(approvals[['endTime']], timezone)
+    approvals[['description']] <- paste("Approval:", approvals[['description']])
+    colors <- c()
     
-    ytop <- vector(mode = "numeric", length = len_data)
-    ybottom <- ytop
+    labels <- format(dateSeq, "%m/%Y")
     
-    ytop[] <- startH
-    ybottom[] <- startH - height
+    for (i in 1:length(approvals[['level']])) {
+      #Assign proper approval colors
+      colors[[i]] <- switch(as.character(approvals[['level']][[i]]),
+                            "2" = "#228B22",
+                            "1" = "#FFD700",
+                            "0" = "#DC143C"
+      )
+    }
     
-    if(!is.null(overlapInfo[[d]])){
+    returnData <- list(
+      startDates = approvals[['startTime']],
+      endDates = approvals[['endTime']],
+      type = approvals[['description']],
+      colors = colors,
+      approvalLabel = labels
+    )
+  }
+  
+  return(returnData)
+}
 
-      newLine_i <- overlapInfo[[d]]$rectToShift
-      line <- overlapInfo[[d]]$lineNum
+#' Parse CORR Thresholds
+#'
+#' @description Retrieves and formats thresholds for the CORR report
+#' @param reportObject The full report JSON object
+#' @param timezone The timezone to parse data into
+parseCorrThresholds <- function(reportObject, timezone){
+  thresholdData <- tryCatch({
+    readThresholds(reportObject)
+  }, error=function(e) {
+    warning(paste("Returning NULL for threhsolds. Error:", e))
+    return(NULL)
+  })
+  
+  formattedData <- list()
+  returnData <- list()
+
+  if(!isEmptyOrBlank(thresholdData) && nrow(thresholdData) > 0){
+    for(i in seq(nrow(thresholdData))){
+      threshold <- thresholdData[i,]
       
-      ytop[newLine_i] <- startH - height*(line - 1) #num lines away from first line
-      ybottom[newLine_i] <- ytop[newLine_i] - height
+      periods <- threshold[['periods']][[1]]
+      type <- rep(threshold[['type']], times=nrow(periods))
+      startDates <- periods[['startTime']]
+      endDates <- periods[['endTime']]
+      value <- periods[['referenceValue']]
+      suppressData <- periods[['suppressData']]
       
+      formattedData <- rbind(formattedData, data.frame(type, startDates, endDates, value, suppressData))
     }
     
-    allData[[d]][['ytop']] <- ytop
-    allData[[d]][['ybottom']] <- ybottom
+    formattedData <- formattedData[which(formattedData[['suppressData']]),]
```

Comment:
That's what I thought as well but it was originally implemented to include only suppressed data and exclude data that wasn't suppressed so I kept the same behavior. I'm thinking "suppress data" must mean something other than what it seems to mean at face value?